### ISSUE TITLE
allow wrapper to set env while sub-invoking other wrappers

### ISSUE THREAD

**Issue by Niraj-Kamdar** [Web3-API/prototype#828]

**Comment by dOrgJelli** [comment]
In order to decide if we should implement this or not, can we please come up with some use-cases where this would come in handy?

I want to make sure this is something that is necessary for wrapper developers, and not something that should all be done by the application developer.

One bad scenario I see with this is upgradeability, and protecting against breaking changes. Take this example:
- The app `Blue` uses wrapper `Foo`
- `Foo` depends on wrapper `Bar`
  - `Blue` -> `Foo` -> `Bar`
- `Foo` configures `Bar`'s environment
- `Blue` does not configure `Bar`'s environment
- If there is no "pre-check" within `Blue`'s app to ensure that all environments are pre-configured, this setup will work, but can fail in the future...
- (Failure Mode) If `Foo` has a version upgrade published where its underlying wasm is changed to NOT include the environment configuration, then `Blue` will become broken the next time it runs.
One downside to implementing this, is it may lead to some wrapper developers configuring wrapper environments "on behalf of" the application developer. If the client does not pre-check the existence of wrapper environments,

**Comment by dOrgJelli** [comment]
TL;DR: I have some concerns :P

**Comment by Niraj-Kamdar** [comment]
I understand your concerns but this will be really needed to change or override the env set by the client from the wrapper. 
Consider a case where ethereum plugin is set with polygon config env from the client but for ens lookup we need mainnet config and need to override the config set by client. If we don't then ENS resolver will failed to fetch wrapper IPFS URI. 

- ENS resolver plugin - requires Ethereum plugin with mainnet config
- App developer - requires Ethereum plugin with polygon config (Ex: wrapper for quickswap)

This is one of many cases that can happen. Another use-case would be aggregator wrapper where we would like to aggregate data from the different sources which can be configured using envs. 

Ex: There can be an aggregator which queries the balance of different chains using the same wrapper with different ENV configs. Let's call it BalanceAggregator and have a wrapper to get balances of a particular chain named BalanceResolver. 
Here's the ENV that can help with this.

- **BalanceAggregator ENV** 
```graphql
type Env {
  polygon: String! # node rpc URL for polygon
  ethereum: String!  # node rpc URL for ethereum
  avalanche: String!
  gnosis: String!
}
```
- **BalanceResolver ENV**
```graphql
type Env {
  rpcUrl: String!
}
```
